### PR TITLE
Pre-processing: Resize area kernels dispatching refactoring.

### PR DESCRIPTION
**Pull request by anna-khakimova**
### Details:
 - *Resize 8U Area refactoring*
 - *Resize 32F Area refactoring*

### CODE REVIEW

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 1]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 3>{}, nullptr);
-        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
-            std::array<uint8_t*, 3> outs = { out1.OutLineB(i), out2.OutLineB(i),
-                                            out3.OutLineB(i) };
-            rowFunc(in.InLineB(i), outs, in.length());
-        }
-    }
-};
+    static inline Unit map(double ratio, int start, int max, int outCoord) {
+        int s = cvFloor(outCoord*ratio);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:814]: `const int s`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 2]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+        float f = static_cast<float>((outCoord+1) - (s+1)/ratio);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:815]: And `single letter variable`. It is difficult for readability.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 3]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-                               const float beta[], float tmp[], const Size& inSz,
-                               const Size& outSz, const int lpi, const int length, const int depth) {
-    const auto rowFunc = type_dispatch<TL>(depth, cv_type_id{}, KT{}, nullptr);
+    std::vector<Unit> xmaps(outSz.width);
+    int  maxdif = 0;
 
-    GAPI_DbgAssert(rowFunc);
+    for (int w = 0; w < outSz.width; w++) {
+        Unit map = mapper.map(w);
+        xmaps[w] = map;
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:875]: 870 `xmaps.reserve(outSz.width)`

874 `xmaps.emplace_back(mapper.map(w))`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 4]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+            maxdif = dif;
+    }
 
-template<typename T, class Mapper, typename KT, typename LT>
-static inline void calcRowLinear(const cv::gapi::fluid::View& in,
-                                 cv::gapi::fluid::Buffer& out,
-                                 cv::gapi::fluid::Buffer& scratch) {
-    GAPI_DbgAssert(is_cv_type_in_list<LT>(out.meta().depth));
+    // This assertion is critical for our trick with chunk sizes:
+    // we would expand a chunk it is is smaller than maximal size
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:883]: `it is is` ->` it is`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 5]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+    // pack the input chunks positions and coefficients into scratch-buffer,
+    // along with the maximal size of chunk (note that chunk size may vary)
 
-    auto inY = in.y();
-    int length = out.length();
-    int outY = out.y();
-    int lpi = out.lpi();
-    GAPI_DbgAssert(outY + lpi <= outSz.height);
+    size_t scratch_bytes =               sizeof(int)
+                         + outSz.width * sizeof(index_type)
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:890]: Alignment

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 6]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+                         + outSz.width * sizeof(alpha_type) * maxdif
+                         +  inSz.width * sizeof(alpha_type);
+    Size scratch_size{static_cast<int>(scratch_bytes), 1};
 
-    GAPI_DbgAssert(lpi <= 4);
+    cv::GMatDesc desc;
+    desc.chan = 1;
+    desc.depth = CV_8UC1;
+    desc.size = scratch_size;
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:898]: `cv::GMatDesc desc(CV_8UC1, 1, scratch_size);`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 7]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-    const auto* alpha = scr.alpha;
-    const auto* clone = scr.clone;
-    const auto* mapsx = scr.mapsx;
-    const auto* beta0 = scr.beta;
-    const auto* mapsy = scr.mapsy;
-    auto* tmp = scr.tmp;
+    auto *maxdf =  scratch.OutLine<int>();
+    auto *index = reinterpret_cast<index_type*>(maxdf + 1);
+    auto *alpha = reinterpret_cast<alpha_type*>(index + outSz.width);
+//  auto *vbuf  = reinterpret_cast<work_type *>(alpha + outSz.width * maxdif);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:906]: Unused code

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 8]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-    const T* src1[4];
-    T* dst[4];
+    for (int w = 0; w < outSz.width; w++) {
+        // adjust input indices so that:
+        // - data chunk is exactly maxdif pixels
+        // - data chunk fits inside input width
+        int index0 = xmaps[w].index0;
+        int index1 = xmaps[w].index1;
+        int i0 = index0, i1 = index1;
+        i1 = (std::min)(i0 + maxdif, in.size.width);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:915]: How does it work? Can it be `std::min(i0 + maxdif, in.size.width);` instead?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 9]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-    const T* src0[4];
-    const T* src1[4];
-    T* dst[4];
+    for (int w = 0; w < outSz.width; w++) {
+        // adjust input indices so that:
+        // - data chunk is exactly maxdif pixels
+        // - data chunk fits inside input width
+        int index0 = xmaps[w].index0;
+        int index1 = xmaps[w].index1;
+        int i0 = index0, i1 = index1;
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:914]: `int i0 = index0, i1 = index1;` -->

```
int i0 = index0;
int i1 = index1;
```

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 10]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+    int    inSz, outSz;
+    double ratio, inv_ratio;
 
-        int buf_width = out.length();
-        GAPI_DbgAssert(in_u.length() == in_v.length());
+    alpha_type  alpha;  // == inv_ratio, rounded
 
-        const auto rowFunc = type_dispatch<i420_to_rgb_supported_types>(out.meta().depth, cv_type_id{}, typed_i420_to_rgb_row<isa_tag_t>{}, nullptr);
+    void init(int _inSz, int _outSz) {
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:778]: Could it be a part of constructor? to keep object consistent straight after creation

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 11]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-        GAPI_DbgAssert(2 == in.meta().chan);
-        GAPI_DbgAssert(1 == out1.meta().chan);
-        GAPI_DbgAssert(1 == out2.meta().chan);
-        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
-        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
-        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+namespace areaDownscale32f {
+struct Mapper: public AreaDownMapper<float, int, float> {
+    Mapper(int _inSz, int _outSz) {
+        init(_inSz, _outSz);
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:792]: looks like we can avoid `init` and just call `AreaDownMapper` constuctor

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 12]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+    int  maxdif = 0;
 
-    GAPI_DbgAssert(rowFunc);
+    for (int w = 0; w < outSz.width; w++) {
+        Unit map = mapper.map(w);
+        xmaps[w] = map;
 
-    rowFunc(dst, src0, src1, alpha, mapsx, beta, inSz, outSz, lpi, length);
-}
+        int dif = map.index1 - map.index0;
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:877]: the more shorter version can be:

     maxdif = std::max(maxdif, map.index1 - map.index0);

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 8 above)
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:915]: why std::min in brackets?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 13]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-};
+        int sx1 = static_cast<int>(ceil(fsx1));
+        int sx2 = static_cast<int>(floor(fsx2));
 
-GAPI_FLUID_KERNEL(FScalePlane32f, ScalePlane32f, true) {
-    static const int Window = 1;
-    static const int LPI = 4;
-    static const auto Kind = cv::GFluidKernel::Kind::Resize;
+        sx2 = (std::min)(sx2, ssize - 1);
+        sx1 = (std::min)(sx1, sx2);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:959]: How does it work?)

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 14]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
 
-    static void initScratch(const cv::GMatDesc & in,
-                            Size outSz, int /*interp*/,
-                            cv::gapi::fluid::Buffer & scratch) {
-        GAPI_DbgAssert(in.depth == CV_32F && in.chan == 1);
+        if (sx1 - fsx1 > threshold) {
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:961]: `(sx1 - fsx1) > threshold`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 15]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+        }
 
-        initScratchLinear<float, linear32f::Mapper>(in, outSz, scratch, 0);
-    }
+        for (int sx = sx1; sx < sx2; sx++) {
+            count++;
+        }
 
-    static void resetScratch(cv::gapi::fluid::Buffer& /*scratch*/) {
+        if (fsx2 - sx2 > threshold) {
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:969]: Please add brackets for `fsx2 - sx2`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 16]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+            count++;
+        }
+        max_count = (std::max)(max_count, count);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:972]: I wondering why is not `std::max(max_count, count);`?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 17]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
+                                                            nullptr);
+    GAPI_DbgAssert(rowFunc);
 
-    static void run(const cv::gapi::fluid::View& in, Size /*sz*/, int /*interp*/,
-                    cv::gapi::fluid::Buffer& out, cv::gapi::fluid::Buffer &scratch) {
-        calcRowLinear<float, areaUpscale32f::Mapper>(in, out, scratch);
+    for (int l = 0; l < lpi; l++) {
+        Unit ymap = ymapper.map(y + l);
+
+        GAPI_Assert(ymap.index1 - ymap.index0 <= 32);
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1780]: could you please give a name to 32 constant?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 18]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
+        return buffer_size;
+    };
 
-namespace areaDownscale8u {
-struct Mapper: public AreaDownMapper<Q0_16, short, Q8_8> {
-    Mapper(int _inSz, int _outSz) {
-        init(_inSz, _outSz);
-    }
-};
+    return resize_area_u8_downscale_sse_buffer_size();
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1087]: Why? Does the lambda make sense?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 19]
```diff
@@ -2058,41 +1832,136 @@ GAPI_FLUID_KERNEL(FScalePlaneArea8u, ScalePlaneArea8u, true) {
...
+                    return UpscalePlaneArea8u::on(in, szOut, interp);
+                else
+                    return   ScalePlaneArea8u::on(in, szOut, interp);
+            }
+            if (CV_32FC1 == type) {
+                if (upscale)
+                    return UpscalePlaneArea32f::on(in, szOut, interp);
+                else
+                    return   ScalePlaneArea32f::on(in, szOut, interp);
+            }
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1940]: Do we need ASSERT here if both conditions are false?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 20]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-        xmaps[w] = map;
-
-        int dif = map.index1 - map.index0;
-        if (dif > maxdif)
-            maxdif = dif;
-    }
+// buffer-fulfill is taken from: ie_preprocess_data_sse42.cpp
+static void initScratchArea_CVKL_U8(const cv::GMatDesc & in,
+                                    const       Size   & outSz,
+                               cv::gapi::fluid::Buffer & scratch) {
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1093]: Alignment

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 21]
```diff
@@ -2058,41 +1832,136 @@ GAPI_FLUID_KERNEL(FScalePlaneArea8u, ScalePlaneArea8u, true) {
...
+        if (cv::INTER_LINEAR == interp) {
+            if (CV_8UC1 == type) {
+                return ScalePlane8u::on(in, szOut, interp);
+            }
+            if (CV_32FC1 == type) {
+                return ScalePlane32f::on(in, szOut, interp);
+            }
+        }
+
+        GAPI_Assert(!"unsupported parameters");
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1952]: can we describe them?

**Review by sivanov-work — APPROVED** [review]

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 22]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
 
-        // start input chunk with adjusted position
-        index[w] = i0;
-    }
+        auto* maxdif = scratch.OutLine<int>();
+        auto* xsi = reinterpret_cast<uint16_t*>(maxdif + 2);
+        auto* ysi = xsi + dwidth;
+        auto* xalpha = ysi + dheight;
+        auto* yalpha = xalpha + dwidth*x_max_count + 8*16;
+    //  auto* vert_sum = yalpha + dheight*y_max_count;
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1142]: Unused code.

**Review by mpashchenkov — COMMENTED** [review]
+- half

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 23]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
+# pragma GCC diagnostic pop
+#endif
 
 GAPI_FLUID_KERNEL(FScalePlaneArea32f, ScalePlaneArea32f, true) {
     static const int Window = 1;
     static const int LPI = 4;
     static const auto Kind = cv::GFluidKernel::Kind::Resize;
 
-    static void initScratch(const cv::GMatDesc& in,
+    static void initScratch(const cv::GMatDesc & in,
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1803]: Please align this spaces and & with file's code style.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 24]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
-    static void resetScratch(cv::gapi::fluid::Buffer& /*scratch*/) {
-    }
+    Size inSz  =  in.meta().size;
+    Size outSz = out.meta().size;
+
+    // this method is valid only for down-scale
+    GAPI_DbgAssert(inSz.width  >= outSz.width);
+    GAPI_DbgAssert(inSz.height >= outSz.height);
+
+//  Mapper xmapper(inSz.width,  outSz.width);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1753]: Debug code?

**Review by mpashchenkov — APPROVED** [review]

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 10 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 11 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 12 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 8 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 17 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 19 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1937]: Assert already exists at the bottom.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 25]
```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
...
-                    cv::gapi::fluid::Buffer & out) {
-        GAPI_DbgAssert(is_cv_type_in_list<nv12_to_rgb_supported_types>(out.meta().depth));
+        double alpha0 =   (index0 + 1 - inCoord0) * inv_ratio;
+        double alpha1 = - (index1 - 1 - inCoord1) * inv_ratio;
 
-        const uchar* uv_row = in_uv.InLineB(0);
-        const uchar* y_rows[2] = { in_y.InLineB(0), in_y.InLineB(1) };
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+        GAPI_Assert(0 <= outCoord && outCoord <= outSz-1);
+        GAPI_Assert(0 <= index0 && index0 < index1 && index1 <= inSz);
```
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 10 above)
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 1 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 2 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 3 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:871]: Prefer to leave as is.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 4 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 5 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:884]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 6 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 7 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 8 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Applied

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 9 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 13 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Changed.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 14 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 15 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 16 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 20 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 22 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Removed.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 26]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
-    *maxdf = maxdif;
-}
+        maxdif[0] = x_max_count;
+        maxdif[1] = y_max_count;
 
-#if defined __GNUC__
-# pragma GCC diagnostic push
-# pragma GCC diagnostic ignored "-Wstrict-aliasing"
-#endif
+        computeResizeAreaTab(src_go_x, dst_go_x, src_full_width,   dwidth, scale_x, xsi, xalpha, x_max_count);
```
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 27]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
+        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out3.meta().depth);
+
+        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+
+        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 3>{}, nullptr);
+        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
+            std::array<uint8_t*, 3> outs = { out1.OutLineB(i), out2.OutLineB(i),
+                                            out3.OutLineB(i) };
```
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 28]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
+        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out3.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out4.meta().depth);
+        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+
+        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 4>{}, nullptr);
+        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
+            std::array<uint8_t*, 4> outs = { out1.OutLineB(i), out2.OutLineB(i),
+                                            out3.OutLineB(i), out4.OutLineB(i) };
```
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 29]
```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
...
+};
 
-    auto* xsi = reinterpret_cast<uint16_t*>(maxdif + 2);
-    auto* ysi    = xsi + dwidth;
-    auto* xalpha = ysi + dheight;
-    auto* yalpha = xalpha + dwidth*x_max_count + 8*16;
-    auto* vert_sum = yalpha + dheight*y_max_count;
+template<typename T, class Mapper, int chs>
+static inline void calcRowLinearC(const cv::gapi::fluid::View& in,
+                           std::array<std::reference_wrapper<cv::gapi::fluid::Buffer>, chs>& out,
```
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 23 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Done

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 24 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp]: Removed

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp (hunk 19 above)
Comment (sivanov-work) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:1937]: ok, i see.

**Pull request merged**